
Running tests... 
----------------------------------------------------------------------
 test_game_controls (GameTest.GameTest.test_game_controls) ... OK (8.627472)s
 test_game_score (GameTest.GameTest.test_game_score) ... OK (15.924771)s
 test_start_game (GameTest.GameTest.test_start_game) ... OK (7.741935)s
 test_verify_garbage_collector_color (GameTest.GameTest.test_verify_garbage_collector_color) ... ERROR (5.537476)s
 test_versus_gameplay (GameTest.GameTest.test_versus_gameplay) ... ERROR (17.528220)s
 test_leaderboard_update (UpdateTest.UpdateTest.test_leaderboard_update) ... OK (32.320238)s
 test_watch_realtime (UpdateTest.UpdateTest.test_watch_realtime) ... OK (12.510453)s
 test_account_creation (UserAccountTest.UserAccountTest.test_account_creation) ... OK (5.201076)s
 test_account_login (UserAccountTest.UserAccountTest.test_account_login) ... OK (4.382728)s
 test_account_logout (UserAccountTest.UserAccountTest.test_account_logout) ... OK (5.242221)s
 test_click_specific_color (UserAccountTest.UserAccountTest.test_click_specific_color) ... ERROR (3.463031)s
 test_send_message (UserAccountTest.UserAccountTest.test_send_message) ... ERROR (18.147383)s
 test_settings_and_play_with_movement (UserAccountTest.UserAccountTest.test_settings_and_play_with_movement) ... ERROR (3.414605)s

======================================================================
ERROR [5.537476s]: GameTest.GameTest.test_verify_garbage_collector_color
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\application\backend\test\GameTest.py", line 215, in test_verify_garbage_collector_color
    lives_element = self.wait.until(EC.presence_of_element_located(
                    ^^^^^^^^^
AttributeError: 'GameTest' object has no attribute 'wait'

======================================================================
ERROR [17.528220s]: GameTest.GameTest.test_versus_gameplay
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\application\backend\test\GameTest.py", line 163, in test_versus_gameplay
    webdriverwait1.until(EC.presence_of_element_located((By.ID, "create-game-button"))).click()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\.venv\Lib\site-packages\selenium\webdriver\support\wait.py", line 105, in until
    raise TimeoutException(message, screen, stacktrace)
selenium.common.exceptions.TimeoutException: Message: 
Stacktrace:
	GetHandleVerifier [0x00007FF63BA46CF5+28821]
	(No symbol) [0x00007FF63B9B3880]
	(No symbol) [0x00007FF63B85578A]
	(No symbol) [0x00007FF63B8A91BE]
	(No symbol) [0x00007FF63B8A94AC]
	(No symbol) [0x00007FF63B8F2647]
	(No symbol) [0x00007FF63B8CF33F]
	(No symbol) [0x00007FF63B8EF412]
	(No symbol) [0x00007FF63B8CF0A3]
	(No symbol) [0x00007FF63B89A778]
	(No symbol) [0x00007FF63B89B8E1]
	GetHandleVerifier [0x00007FF63BD7FCED+3408013]
	GetHandleVerifier [0x00007FF63BD9745F+3504127]
	GetHandleVerifier [0x00007FF63BD8B63D+3455453]
	GetHandleVerifier [0x00007FF63BB0BDFB+835995]
	(No symbol) [0x00007FF63B9BEB9F]
	(No symbol) [0x00007FF63B9BA854]
	(No symbol) [0x00007FF63B9BA9ED]
	(No symbol) [0x00007FF63B9AA1D9]
	BaseThreadInitThunk [0x00007FFDEE58259D+29]
	RtlUserThreadStart [0x00007FFDEFECAF38+40]


======================================================================
ERROR [3.463031s]: UserAccountTest.UserAccountTest.test_click_specific_color
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\application\backend\test\UserAccountTest.py", line 189, in test_click_specific_color
    self.login()
    ^^^^^^^^^^
AttributeError: 'UserAccountTest' object has no attribute 'login'

======================================================================
ERROR [18.147383s]: UserAccountTest.UserAccountTest.test_send_message
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\application\backend\test\UserAccountTest.py", line 167, in test_send_message
    received_message = webdriverwait2.until(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\.venv\Lib\site-packages\selenium\webdriver\support\wait.py", line 105, in until
    raise TimeoutException(message, screen, stacktrace)
selenium.common.exceptions.TimeoutException: Message: 
Stacktrace:
	GetHandleVerifier [0x00007FF63BA46CF5+28821]
	(No symbol) [0x00007FF63B9B3880]
	(No symbol) [0x00007FF63B85578A]
	(No symbol) [0x00007FF63B8A91BE]
	(No symbol) [0x00007FF63B8A94AC]
	(No symbol) [0x00007FF63B8F2647]
	(No symbol) [0x00007FF63B8CF33F]
	(No symbol) [0x00007FF63B8EF412]
	(No symbol) [0x00007FF63B8CF0A3]
	(No symbol) [0x00007FF63B89A778]
	(No symbol) [0x00007FF63B89B8E1]
	GetHandleVerifier [0x00007FF63BD7FCED+3408013]
	GetHandleVerifier [0x00007FF63BD9745F+3504127]
	GetHandleVerifier [0x00007FF63BD8B63D+3455453]
	GetHandleVerifier [0x00007FF63BB0BDFB+835995]
	(No symbol) [0x00007FF63B9BEB9F]
	(No symbol) [0x00007FF63B9BA854]
	(No symbol) [0x00007FF63B9BA9ED]
	(No symbol) [0x00007FF63B9AA1D9]
	BaseThreadInitThunk [0x00007FFDEE58259D+29]
	RtlUserThreadStart [0x00007FFDEFECAF38+40]


======================================================================
ERROR [3.414605s]: UserAccountTest.UserAccountTest.test_settings_and_play_with_movement
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\Users\laxjo\OneDrive\Desktop\Capstone\aldwairi-projects-danglingpointers\application\backend\test\UserAccountTest.py", line 229, in test_settings_and_play_with_movement
    self.login_user()
    ^^^^^^^^^^^^^^^
AttributeError: 'UserAccountTest' object has no attribute 'login_user'

----------------------------------------------------------------------
Ran 13 tests in 0:02:52

FAILED
 (Errors=5)

Generating HTML reports... 
reports\TestResults_GameTest_UpdateTest_UserAccountTest_2024-12-07_16-19-38.html
